ddtrace/_logger.py

```diff
@@ -0,0 +1,59 @@
+import logging
+from logging.handlers import RotatingFileHandler
+import os
+
+from ddtrace.internal.utils.formats import asbool
+
+
+DEFAULT_FILE_SIZE_BYTES = 15 << 20  # 15 MB
+
+
+def configure_ddtrace_logger():
+    # type: () -> None
+    """Configures ddtrace log levels and file paths.
+
+    Customization is possible with the environment variables:
+        ``DD_TRACE_DEBUG``, ``DD_TRACE_LOG_FILE_LEVEL``, and ``DD_TRACE_LOG_FILE``
+
+    By default, when none of the settings have been changed, ddtrace loggers
+        inherit from the root logger in the logging module and no logs are written to a file.
+
+    When DD_TRACE_DEBUG has been enabled:
+        - Propagate logs up so logs show up in the application logs if a file path wasn't provided
+        - Logs are routed to a file when DD_TRACE_LOG_FILE is specified, using the log level in DD_TRACE_LOG_FILE_LEVEL.
+        - Child loggers inherit from the parent ddtrace logger
+
+    Note: The ddtrace-run logs under commands/ddtrace-run do not follow DD_TRACE_LOG_FILE
+        if DD_TRACE_DEBUG is enabled.
+
+    """
+    debug_enabled = asbool(os.environ.get("DD_TRACE_DEBUG", "false"))
+    log_file_level = os.environ.get("DD_TRACE_LOG_FILE_LEVEL", "DEBUG")
+    log_path = os.environ.get("DD_TRACE_LOG_FILE")
+    max_file_bytes = int(os.environ.get("DD_TRACE_FILE_SIZE_BYTES", DEFAULT_FILE_SIZE_BYTES))
+    num_backup = 1
+    log_format = "%(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] - %(message)s"
+    log_formatter = logging.Formatter(log_format)
+
+    ddtrace_logger = logging.getLogger("ddtrace")
+
+    if debug_enabled is True:
+        ddtrace_logger.setLevel(logging.DEBUG)
+        ddtrace_logger.debug("Debug mode has been enabled for the ddtrace logger.")
+        file_log_level_value = logging.getLevelName("DEBUG")
+
+    if "Level" not in str(logging.getLevelName(log_file_level)):  # Result starts with 'Level' if no matches
+        file_log_level_value = logging.getLevelName(log_file_level)
+    else:
+        file_log_level_value = logging.getLevelName("DEBUG")
```

Comment:
```suggestion

    file_log_level_value = getattr(logging, log_file_level.upper())
```

We have the string representation but we need the numeric value. `getLevelName` will still only get us a string representation where we want the number value, e.g. `logging.DEBUG`.

There isn't an awesome way to get this from `logging` (that I've found yet), so the typical approach is ☝🏻 

We also don't need to set the log level value `if debug_enabled:` since the default value is `"DEBUG"`.

If we use this `getattr` approach, we can actually move it up under L31

Reply:
🤔 I worry since they mention in the docs that they broke that behavior, but added it back if they will ever decide in the future to remove it again.

if you want to keep `getLevelName`, just make sure to `log_file_level.upper()` `getLevelName("debug")` doesn't work.

I wonder how we should warn about an unknown level provided?

e.g. with `getattr(logging, "UNKNOWN")` you'll get:

```
>>> getattr(logging, "UNKNOWN")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'logging' has no attribute 'UNKNOWN'
```

```
>>> root.setLevel(logging.getLevelName("UNKNOWN"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/brett.langdon/.pyenv/versions/3.10.2/lib/python3.10/logging/__init__.py", line 1452, in setLevel
    self.level = _checkLevel(level)
  File "/Users/brett.langdon/.pyenv/versions/3.10.2/lib/python3.10/logging/__init__.py", line 198, in _checkLevel
    raise ValueError("Unknown level: %r" % level)
ValueError: Unknown level: 'Level UNKNOWN'
```

Which really isn't bad, other than the "Level " part.

Or we could do:

```python
level_value = logging.getLevelName(log_file_level.upper())
if not isinstance(level_value, int):
    raise ValueError("Unknown logging level: {!r}".format(log_file_level))
```

Which is also good... lots of options. As long as the error message when an unknown name is provided is good. Also preference to raising an exception over logging.

Reply:
That makes sense, I can raise an error with `getattr` too , so I'll update it with that and see if that might look better than the `getLeveLName` route just to make it more future proof. The exception part makes sense to me, since it's unclear what a log level should be if something like "UNKNOWN" was used.

Reply:
> The exception part makes sense to me, since it's unclear what a log level should be if something like "UNKNOWN" was used.

👍🏻 yup, we don't want to take an input from the user and just "ignore" it if an invalid value was provided, much better for us to raise an exception to force them to address the issue.